auto-connect from default connection in `.ini` file

edublancas opened the pull request:
## Describe your changes

when initializing the extension via `%load_ext sql`, JupySQL will automatically connect to a database if the configuration file has a database in a `[default]` section

## Issue number

Closes #X

## Checklist before requesting a review

- [ ] Performed a self-review of my code
- [ ] Formatted my code with [`pkgmt format`](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#linting-formatting)
- [ ] Added [tests](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#testing) (when necessary).
- [ ] Added [docstring](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#documenting-changes-and-new-features) documentation and update the [changelog](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#changelog) (when needed)



<!-- readthedocs-preview jupysql start -->
----
:books: Documentation preview :books:: https://jupysql--819.org.readthedocs.build/en/819/

<!-- readthedocs-preview jupysql end -->

neelasha23 commented on src/sql/display.py:
```diff
@@ -92,6 +92,16 @@ def message_success(message):
+    display(
+        Message(
+            message,
+            style="background-color:#fff3cd;color:#d39e00",
```
is there any convention that we are following for these message styles, e.g., when should we use which style and how do we know which one to use for any future usecase?

edublancas replied:
there's not explicit convention because this is subjective but what I do is show a message when a user operation succeeded and a warning when something went wrong but we can still proceed

neelasha23 commented on doc/connecting.md:
````diff
@@ -20,26 +20,54 @@ myst:
+```{versionchanged} 0.10.0
+```
+
+Using a `.ini` file is the recommended way to connect to databases. By default, JupySQL reads the `~/.jupysql/connections.ini` file, but you can change this setting. A `.ini` file looks like this:
````
Shouldn't it be looking in the path where `jupysql` folder actually exists? e.g., in my machine the path doesn't exist:
```
(jupysql) neelashasen@192 jupysql % ls /Users/neelashasen/.jupysql
ls: /Users/neelashasen/.jupysql: No such file or directory
```

<img width="732" alt="Screenshot 2023-08-20 at 12 07 14 PM" src="https://github.com/ploomber/jupysql/assets/29096135/fb82d644-b100-4bbb-98e0-16a735e01021"> 

I went on to create a `connections.ini` file in my jupysql directory:
<img width="367" alt="Screenshot 2023-08-20 at 12 03 30 PM" src="https://github.com/ploomber/jupysql/assets/29096135/79283a91-d3f4-48e4-92fa-082081fb8f3c">

And it throws error:

<img width="1099" alt="Screenshot 2023-08-20 at 12 04 05 PM" src="https://github.com/ploomber/jupysql/assets/29096135/d48daa4e-d68e-4213-a099-f103c624f35a">


Maybe there should be some warnings / errors if `~/.jupysql/connections.ini` is not a valid path?

edublancas replied:
the path doesn't exist, the user has to create it. I don't think we should show warnings because this feature is optional so warning them that the .ini file doesn't exist will be confusing.

I don't understand the error you're reporting. It's likely failing because the file isn't there, this is how it looks on my end:

```
(jupysql) eduardo@macbookair dev/jupysql (dev/default-conn) » cat ~/.jupysql/connections.ini
[default]
database = :memory:
drivername = duckdb

[disk]
drivername = duckdb
database = /Users/eduardo/.jupysql/duck.db

(jupysql) eduardo@macbookair dev/jupysql (dev/default-conn) » ipython
Python 3.10.12 (main, Jul  5 2023, 15:02:25) [Clang 14.0.6 ]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.14.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: %load_ext sql
Found pyproject.toml from '/Users/eduardo/dev/jupysql'

In [2]: %sql --connections
Out[2]:
Active connections:
+---------+--------------------+---------+
| current |        url         |  alias  |
+---------+--------------------+---------+
|    *    | duckdb:///:memory: | default |
+---------+--------------------+---------+

In [3]:
```

neelasha23 commented:
We can add some some message that `default connection loaded from  ~/.jupysql/connections.ini file`:
<img width="674" alt="Screenshot 2023-08-20 at 12 16 58 PM" src="https://github.com/ploomber/jupysql/assets/29096135/b0d09473-8de4-4f62-8753-4c90b5a1b28d">

neelasha23 commented on src/tests/test_config.py:
```diff
@@ -19,3 +45,263 @@ def test_dsn_filename_resolves_user_directory(sql_magic):
+def test_magic_initialization_with_corrupted_pyproject(tmp_empty, ip_no_magics, capsys):
+    Path("pyproject.toml").write_text(
+        """
+[tool.jupysql.SqlMagic]
```
the contents look correct of pyproject. why is this courrupted?

edublancas replied:
the string isn't quoted

neelasha23 commented on CHANGELOG.md:
```diff
@@ -1,6 +1,7 @@
 # CHANGELOG
 
 ## 0.10.1dev
+* [Feature] Automatically connect if the `dsn_filename` contains a `default` section
```
please add that it looks in the default location `~/.jupysql/connections.ini`

edublancas replied:
https://github.com/ploomber/jupysql/pull/819/commits/7c2ec814c8e6b4534de137a46e80a2c4ca6f6b49

edublancas commented:
> We can add some some message that default connection loaded from  ~/.jupysql/connections.ini file:

we're tackling that here: https://github.com/ploomber/jupysql/issues/807

edublancas commented:
> Some of the information looks scattered in the docs. Currently there are two ways of setting the connections file: Either set in the notebook itself %config SqlMagic.dsn_filename = "connections.ini or define it in the pyproject.toml https://jupysql.ploomber.io/en/latest/api/configuration.html#loading-from-pyproject-toml.
I think it might be good if this information can also be added for a quick reference [here](https://jupysql--819.org.readthedocs.build/en/819/connecting.html#connecting-with-a-ini-file)

not sure what your suggestion is. the objective  [here](https://jupysql--819.org.readthedocs.build/en/819/connecting.html#connecting-with-a-ini-file) is to show at a high-level what options the user has. but I modified the connection file tutorial and added a link to the pyproject toml section: https://github.com/ploomber/jupysql/pull/819/commits/d9abd25e827f7417502ecbccf3bca3107f42001f

neelasha23 commented:
> d9abd25

yes this is what I meant!
